### ISSUE TITLE
Istiod does not have prometheus scraping annotations when metrics merging is disabled

### ISSUE THREAD

**Issue by alexbrand** [istio/istio#33280]
**Bug description**
Currently, the metrics merging configuration (`meshConfig.enablePrometheusMerge`) controls whether the Prometheus scraping annotations are placed on istiod pods. This makes it impossible to monitor istiod with Prometheus when metrics merging is disabled.

[Metrics merging](https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-metrics-merging) controls whether application metrics should be merged with istio's metrics. Metrics merging is a data plane concern and should not affect the prometheus-related annotations on control plane pods.

**Affected product area (please put an X in all that apply)**

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Expected behavior**
Prometheus annotations can be added to the istio control plane Pods regardless of whether metrics merging is enabled or disabled.

**Steps to reproduce the bug**
1. Install istio with `meshConfig.enablePrometheusMerge` set to `false`
2. See missing prometheus annotations on control plane Pods (e.g. istiod)

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
1.9.5

**Comment by zhlsunshine** [comment]
Hi @alexbrand, I noticed that [PR#33285](https://github.com/istio/istio/pull/33285) has been closed, so I take this, should I?

**Comment by zhlsunshine** [comment]
Hi @alexbrand, I noticed that [PR#33285](https://github.com/istio/istio/pull/33285) has been closed, so I take this, could I?

**Comment by bianpengyuan** [comment]
> Hi @alexbrand, I noticed that [PR#33285](https://github.com/istio/istio/pull/33285) has been closed, so I take this, could I?

@zhlsunshine what is the planned fix? The PR gets closed because adding those annotations by default will probably cause double scraping. This can be solved by adding prometheus annotations via k8s overlay. Should we just document that here?

**Comment by zhlsunshine** [comment]
@bianpengyuan I just check some code, but no actionable plan for now. However, I think I have missed your discussion before, so my question is what's your means about the double scraping? And I think adding the annotations via k8s overlay would be a solution, but I'm not trying in this way.

**Comment by zhlsunshine** [comment]
@bianpengyuan I just check some code, but no actionable plan for now. However, I think I have missed your discussion before, so my question is what's your means about the double scraping? And I think adding the annotations via k8s overlay would be a solution.

**Comment by zhlsunshine** [comment]
BTW, @bianpengyuan I am still reading code to find the solution

**Comment by bianpengyuan** [comment]
> what's your means about the double scraping?

Meaning people probably have already set up a dedicated istiod scraping job in addition to the general scraping job based on prom annotations. If we add those annotations regardless, prom could scrape istiod twice and all the aggregated data in prom is going to be doubled. Considering this, it is probably easier to just document how do you add prometheus annotations to istiod deployment here or probably at Istio.io

**Comment by zhlsunshine** [comment]
Thanks for your reply @bianpengyuan !  Or there should be a default action for istiod, adding prom annotations by default, then no extra general scraping job, should it be a choice?

**Comment by zhlsunshine** [comment]
Hi @alexbrand , @bianpengyuan I understand that the dedicated istiod scraping job you mentioned above has no any prome annotation, but additional general scraping job works with prome annotations. Am I right? If yes, I think it should be a different issue.

**Comment by bianpengyuan** [comment]
> Thanks for your reply @bianpengyuan ! Or there should be a default action for istiod, adding prom annotations by default, then no extra general scraping job, should it be a choice?

Yeah the problem is that we don't know how folks prometheus is set up, and we don't want user suddenly scrape istiod twice after upgrading.
